src/App.jsx

```diff
@@ -14,70 +15,65 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+  return (
+    <main className="section container">
+      {selectedGood
+        ? (
+          <h1 className="title is-flex is-align-items-center">
+            {`${selectedGood} is selected`}
             <button
-              data-cy="RemoveButton"
+              onClick={() => setSelectedGood('')}
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              className="delete ml-3"
+            />
+          </h1>
+        ) : (
+          <h1 className="title is-flex is-align-items-center">
+            No goods selected
+          </h1>
+        )
+      }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map((good) => {
+            const isSelected = selectedGood === good;
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+            return (
+              <tr
+                key={good}
+                data-cy="Good"
+                className={isSelected && 'has-background-success-light'}
+              >
+                <td>
+                  <button
+                    onClick={
+                      () => setSelectedGood(isSelected ? '' : good)
```

Comment:
```suggestion
                      () => selectGood(isSelected, good)
```
Same here.

![image](https://github.com/mate-academy/react_goods-selector-js/assets/102293862/c9893185-f14a-4435-9b56-600432d4c775)

This way your code will be more straightforward and easier to support and extend in the future.